inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -144,7 +144,7 @@ def _handle_app_source_command(
 
         application_source_manager.add(
             ApplicationAddSourceParameters(
-                file_name=repo_filename,
+                source_list_file_name=repo_filename,
                 gpg_key_name=gpg_key_name,
                 gpg_key_uri=gpg_key_uri,
                 sources=add_source_pkgs,
```

Comment:
**Suggestion:** Extract the logic for parsing source packages into a separate function to avoid code duplication and improve maintainability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                add_source_pkgs = parse_source_packages(parsed_head, "applicationSource/add/repo/repos")
```